### PR TITLE
Added a bank details file

### PR DESCRIPTION
**Pull request by AJAYSINGHBISHT12**
Targets #2989

### Description

added a table in bankdetails

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: app/Models/BankDetails.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class BankDetails extends Model
+{
+    use HasFactory;
+    protected $table = 'bank_details';
+    protected $fillable = ['id', 'key', 'label', 'value'];
```
Comment (GauravGusain98) [inline app/Models/BankDetails.php:12]: simply use guarded function when you want all field to be fillable
```suggestion
    protected $guarded = [];
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

File: database/migrations/2023_04_14_181019_create_bank_details_table.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    public function up()
+    {
+        Schema::create('bank_details', function (Blueprint $table) {
+            $table->unsignedBigInteger('id')->primary();
+            $table->string('key')->nullable(false);
+            $table->string('label')->nullable(false);
+            $table->string('value')->nullable(false);
+            $table->timestamp('created_at')->nullable(false);
+            $table->timestamp('updated_at')->nullable(false);
+        });
```
Comment (Hemrajbharti123) [inline database/migrations/2023_04_14_181019_create_bank_details_table.php:23]: @AJAYSINGHBISHT12 For code optimization we don't use this way created_at and updated_at.

**Review by Hemrajbharti123 — APPROVED** [review]
LGTM

**Comment by AkshitRawat1207** [comment]
Added Bank Details form and submitted data
Screenshot 2023-06-21 123429
Screenshot 2023-06-21 123453

**Comment by GauravGusain98** [comment]
### Testing feedbacks

1. You don't need to show all the data. Most of these fields are not needed to be displayed in UI. In the UI only show the label and value.
image
2. Put the key as the second field. When a user enters the label then automatically populates the key field value as sluggified version of the Label.  Read about slug from [here](https://laravel.com/docs/8.x/helpers#method-str-slug)
image
3. Also provide the functionality to edit existing fields. Provide and edit icon, clicking on this icon should open a modal in which the label, key, and value should be shown. Here we don't want the functionality to update the key automatically when user enter the label
4. Also add backend validation for unique values in key column

**Comment by AkshitRawat1207** [comment]
I have made the requested changes to the code based on your feedback. The script has been updated accordingly. If you have any further questions or need additional assistance, please let me know. I'm here to help!

Screenshot 2023-06-26 181452
Screenshot 2023-06-26 181613
Screenshot 2023-06-26 181604
Screenshot 2023-06-26 181538

File: resources/js/app.js [hunk 3]
```diff
@@ -210,6 +210,18 @@ $(document).ready(() => {
...
+	$('.edit-btn').click(function() {
+		var id = $(this).data('id');
+		var label = $(this).data('label');
+		var value = $(this).data('value');
+
+		$('#editId').val(id);
+		$('#editLabel').val(label);
+		$('#editValue').val(value);
+
+		$('#editModal').modal('show');
```
Comment (GauravGusain98) [inline resources/js/app.js:223]: Can we use more specific id or class names? As this app.js loads on many screens so it is possible that the same class name or id name may be present on those screens. So it's not a good practice to use generic names

File: resources/views/settings/bank-details/index.blade.php [hunk 4]
```diff
@@ -11,24 +11,82 @@
...
+        <div class="modal-dialog" role="document">
+            <div class="modal-content">
+                <div class="modal-header">
+                    <h5 class="modal-title" id="editModalLabel">Edit Bank Detail</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                        <span aria-hidden="true">&times;</span>
+                    </button>
+                </div>
+                <div class="modal-body">
+                    <form action="{{ route('bank-details.update', ['bankDetail' => '__bankDetailId__']) }}" method="POST" id="editBankDetailForm">
```
Comment (GauravGusain98) [inline resources/views/settings/bank-details/index.blade.php:47]: Why you have used `__` underscore in the value? You can simply pass the value.

File: resources/views/settings/bank-details/index.blade.php [hunk 5]
```diff
@@ -11,24 +11,82 @@
...
+
+                $('#editModal').modal('show');
+            });
+            $(document).ready(function() {
+            @if(session('success'))
+                alert('{{ session('success') }}');
+            @endif
+         });
+        });
+    </script>
```
Comment (GauravGusain98) [inline resources/views/settings/bank-details/index.blade.php:91]: Move this to app.js file

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

File: resources/views/settings/bank-details/index.blade.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            </tr>
+        </thead>
+        <tbody>
+            @foreach ($bankDetails as $bankDetail)
+                <tr>
+                    <td>{{ $bankDetail->id }}</td>
+                    <td>{{ $bankDetail->label }}</td>
+                    <td>{{ $bankDetail->value }}</td>
+                    <td>
+                        <a href="#" class="edit-btn" data-id="{{ $bankDetail->id }}" data-label="{{ $bankDetail->label }}" data-value="{{ $bankDetail->value }}">
```
Comment (LavishRawat) [inline resources/views/settings/bank-details/index.blade.php:26]: Remove `` href="#" `` if not required.

File: resources/views/settings/bank-details/create.blade.php [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+            <input type="text" class="form-control" id="value" name="value" required>
+        </div>
+        <div class="text-center col-md-6">
+            <button type="submit" class="btn btn-primary">Submit</button>
+        </div>
+    </form>
+</div>
+
+
+{{-- <script src="https://code.jquery.com/jquery-3.6.0.min.js"></script> --}}
```
Comment (LavishRawat) [inline resources/views/settings/bank-details/create.blade.php:39]: Remove this line if not required.

**Review by LavishRawat — CHANGES_REQUESTED** [review]

**Comment by LavishRawat** [comment]
@AkshitRawat1207 Please resolve the merge conflicts also.

File: resources/js/app.js [hunk 8]
```diff
@@ -211,6 +211,18 @@ $(document).ready(() => {
...
+		var label = $(this).data("label");
+		var value = $(this).data("value");
+
+		$("#editId").val(id);
+		$("#editLabel").val(label);
+		$("#editValue").val(value);
+
+		$("#editModal").modal("show");
+	});
 });
```
Comment (GauravGusain98) [inline resources/js/app.js:226]: Are we using it? If yes, then please use a specific class and id name. The current names are too generic and its possible that someone can use the same name at other places.

So whenever we give name to id or class we try to use the name that is specifically for the functionality such that the chance of other people using it at other places become very low

File: app/Models/BankDetails.php [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class BankDetails extends Model
+{
+    use HasFactory;
+    protected $table = 'bank_details';
+    protected $guarded = [];
+    protected $fillable = ['key', 'label', 'value'];
```
Comment (GauravGusain98) [inline app/Models/BankDetails.php:13]: The `guarded = []` attribute means that all the columns are fillable. You don't need to provide fillable attribute with `guarded`

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]

File: resources/js/app.js (hunk 8 above)
Comment (AkshitRawat1207) [reply resources/js/app.js:226]: no sir we are no using it forgot to remove this code

Comment (GauravGusain98) [reply resources/js/app.js:226]: Ok remove it then

File: app/Models/BankDetails.php (hunk 9 above)
Comment (AkshitRawat1207) [reply app/Models/BankDetails.php:13]: okay sir

**Review by GauravGusain98 — APPROVED** [review]
LGTM

**Pull request merged**